packages/advanced-logic/src/extensions/payment-network/erc20/fee-proxy-contract.ts

```diff
@@ -14,363 +12,194 @@ import * as walletAddressValidator from 'wallet-address-validator';
-        ', ',
-      )}`,
-    );
-  }
+    super(extensionId, currentVersion);
+    this.actions = {
+      ...this.actions,
+      [ExtensionTypes.PnFeeReferenceBased.ACTION.ADD_FEE]: this.applyAddFee.bind(this),
+    };
+  }
+
+  /**
+   * Creates the extensionsData to create the extension ERC20 fee proxy contract payment detection
+   *
+   * @param creationParameters extensions parameters to create
+   *
+   * @returns IExtensionCreationAction the extensionsData to be stored in the request
+   */
+  public createCreationAction(
+    creationParameters: TCreationParameters,
+  ): ExtensionTypes.IAction<TCreationParameters> {
+    if (creationParameters.feeAddress && !this.isValidAddress(creationParameters.feeAddress)) {
+      throw Error('feeAddress is not a valid address');
+    }
 
-  const copiedExtensionState: RequestLogicTypes.IExtensionStates = Utils.deepCopy(extensionsState);
+    if (creationParameters.feeAmount && !Utils.amount.isValid(creationParameters.feeAmount)) {
+      throw Error('feeAmount is not a valid amount');
+    }
 
-  if (extensionAction.action === ExtensionTypes.PnFeeReferenceBased.ACTION.CREATE) {
-    if (requestState.extensions[extensionAction.id]) {
-      throw Error(`This extension has already been created`);
+    if (!creationParameters.feeAmount && creationParameters.feeAddress) {
+      throw Error('feeAmount requires feeAddress');
+    }
+    if (creationParameters.feeAmount && !creationParameters.feeAddress) {
+      throw Error('feeAddress requires feeAmount');
     }
 
-    copiedExtensionState[extensionAction.id] = applyCreation(extensionAction, timestamp);
+    return super.createCreationAction(
+      creationParameters,
+    ) as ExtensionTypes.IAction<TCreationParameters>;
+  }
+
+  /**
+   * Creates the extensionsData to add a fee address
+   *
+   * @param addFeeParameters extensions parameters to create
+   *
+   * @returns IAction the extensionsData to be stored in the request
+   */
+  public createAddFeeAction(
+    addFeeParameters: ExtensionTypes.PnFeeReferenceBased.IAddFeeParameters,
+  ): ExtensionTypes.IAction {
+    if (addFeeParameters.feeAddress && !this.isValidAddress(addFeeParameters.feeAddress)) {
+      throw Error('feeAddress is not a valid address');
+    }
 
-    return copiedExtensionState;
-  }
+    if (addFeeParameters.feeAmount && !Utils.amount.isValid(addFeeParameters.feeAmount)) {
+      throw Error('feeAmount is not a valid amount');
+    }
 
-  // if the action is not "create", the state must have been created before
-  if (!requestState.extensions[extensionAction.id]) {
-    throw Error(`The extension should be created before receiving any other action`);
-  }
+    if (!addFeeParameters.feeAmount && addFeeParameters.feeAddress) {
+      throw Error('feeAmount requires feeAddress');
+    }
+    if (addFeeParameters.feeAmount && !addFeeParameters.feeAddress) {
+      throw Error('feeAddress requires feeAmount');
+    }
```

Comment:
Main comment: the two error labels are mistakenly swapped.
Secondary comment: I think it's clearer and more maintanable if the condition follows the same pattern as the error label ("x requires y" => `if (x && !y)`. It may have prevented this error (already met somewhere else in the lib by the way)